### ISSUE TITLE
Integrate with Nextcloud CalDAV API [$185]

### ISSUE THREAD

**Issue by juliushaertl** [nextcloud/deck#15]
This issue is about integrating deck with Nextclouds CalDAV API.

Current plan from the discussion below:

| Deck | CalDAV | Tasks |
|-|-|-|
| Board | `VCALENDAR` | List |
| Stack | `VTODO` without `RELATED-TO` property | Root-Task |
| Card | `VTODO` with `RELATED-TO` (UUID of the Root-Task / Stack) | Subtask |
| Tags | [`CATEGORIES`](https://tools.ietf.org/html/rfc5545#section-3.8.1.2) or custom implementation (Nextcloud tagging system) | Currently uses `CATEGORIES`, integration with Nextcloud tags would be better, see https://github.com/nextcloud/tasks/issues/89 |
| Assigned users | [`ATTENDEE`](https://tools.ietf.org/html/rfc5545#section-3.8.4.1) or custom implementation | [`ATTENDEE`](https://tools.ietf.org/html/rfc5545#section-3.8.4.1) or custom implementation, see https://github.com/nextcloud/tasks/issues/72 |
| Recurring stacks & cards | [`RRULE`](https://tools.ietf.org/html/rfc5545#section-3.8.5.3) | Recurring tasks & subtasks, see https://github.com/nextcloud/tasks/issues/34 |
| Attachments | [`ATTACH`](https://tools.ietf.org/html/rfc5545#section-3.8.1.1) or custom implementation | [`ATTACH`](https://tools.ietf.org/html/rfc5545#section-3.8.1.1) or custom implementation, see https://github.com/nextcloud/tasks/issues/91 |

> Tags, assigning users, attachments and recurrence is not supported by Tasks yet either, but it would make much sense to implement these features. And if we see that it is not really possible to implement this with pure CalDAV I would be open to implement it without syncing it via CalDAV and have it only available in the webinterface. This would still be better than not having it at all in Tasks and would not limit Deck.

1. Expose deck boards as VCALENDAR, try to map as much data from deck to the VTODO (read only access) #1545 
2. Allow updating VCALENDAR entries, try to map as much as possible from vtodo to the deck database and store the rest as plain vtodo data, so we can still push it out over CalDAV (write access) #2399
3. Deck data could be moved to the DAV backend.

**Comment by joergmschulz** [comment]
as sharing the cards is essential, a log of changes to 
# Activities 
would be nice.

**Comment by poVoq** [comment]
XMPP group-chat notifications for activities would be also nice.

**Comment by poVoq** [comment]
Would it be possible to have a common due date that is linked with the calendar as a shared event between multiple cards as a "milestone" with it's own colored label displayed on the card? I.e. some sort of way to link labels to CalDav events?

**Comment by poVoq** [comment]
Another possible integration would be to link cards to CalDAV task categories, so that if a user creates a new CalDAV task in an external application (lets say OpenTasks on Android) and chooses the right category for it, deck automatically uses the name of that category to sort it to the card with the same name.

**Comment by juliushaertl** [comment]
I thought a lot about how to integrate with the Nextcloud WebDAV/CalDAV/CardDAV interfaces, it will be not as simple as i initially thought. I'll take some time next week to write up what i had in mind and publish some kind of roadmap. :wink:

**Comment by simonpostma** [comment]
If subtasks become a thing in Deck, it can easily compete with whatever kanban system out there. Keeping the syncing with Tasks optional would be great imo. 

Perhaps a functionality to make multiple decks if that is possible?

**Comment by 9662** [comment]
> XMPP group-chat notifications for activities would be also nice.

It is a reasonable idea, but it belongs to the notifications part of Nextcloud, not to Deck.

> Would it be possible to have a common due date that is linked with the calendar as a shared event between multiple cards as a "milestone" with its own colored label displayed on the card? I.e. some sort of way to link labels to CalDav events?

Reinventing the bug tracker? They're supposed to complement, not duplicate each other. You can probably do this already via webhooks from your bug tracker anyway.

**Comment by poVoq** [comment]
@9662: 
Regarding the first comment: maybe, but you might not want to spam your group-board with all the file etc. changes, while deck and task updates might be more interesting.

Regarding the second: no, what have milestone dates to do with a bug-tacker? I think it is worth to have common event linked date labels on the cards so that it is quickly visible to the users what kind of common deadlines might be related to which cards.

**Comment by juliushaertl** [comment]
https://github.com/nextcloud/server/pull/2829 will allow apps to register custom sabre plugins, if this is in we could create our own CalDAV endpoint at /dav/deck that will allow users to subscribe to within their calendar apps.

Here is a short summary of what I planned for that endpoint:
- A board should exposed as a VCALENDAR
- Cards will be exposed as VTODO
- Due dates can be added and exposed as VEVENT
My first plan was to just add the due dates to existing calendars, but it looks like there is no easy way to do that at the moment. Also there are to many issues with calendar permissions and adding to much complexity, so I think the above proposed solution will be the best way. 

> Another possible integration would be to link cards to CalDAV task categories, so that if a user creates a new CalDAV task in an external application (lets say OpenTasks on Android) and chooses the right category for it, Deck automatically uses the name of that category to sort it to the card with the same name.

@poVoq That should be possible with the above solution.

> Would it be possible to have a common due date that is linked with the calendar as a shared event between multiple cards as a "milestone" with its own colored label displayed on the card? I.e. some sort of way to link labels to CalDav events?

@poVoq This is a far more advanced feature. Let's just focus on the basics for now.

> If subtasks become a thing in Deck, it can easily compete with whatever kanban system out there. Keeping the syncing with Tasks optional would be great imo.

@simonpostm That should not be a problem, if there will be a dedicated dav endpoint for the deck app.

> Perhaps a functionality to make multiple decks if that is possible?

@simonpostma Not sure if I understand that correctly. Can you describe that a bit more?

>> XMPP group-chat notifications for activities would be also nice.

> It is a reasonable idea, but it belongs to the notifications part of Nextcloud, not to Deck.

:+1: 

Hope I didn't forgot to comment on something. :wink:

**Comment by simonpostma** [comment]
@juliushaertl 

I uninstalled Deck because i need my kanban to have more features at this point in time, hasty as i was i didn't remember that Deck already has the feature to make multiple boards. Sorry for that.

**Comment by simonpostma** [comment]
Is there any ETA on the next update? Not trying to pressure you, just can't wait to exchange kanbanflow with Deck

**Comment by juliushaertl** [comment]
Sorry for the delay, I've been quite busy this week. It is on my list for the upcoming week.

**Comment by simonpostma** [comment]
It's clear you're very busy. Do take your time! I'll be looking to donate some to you after it's released.

**Comment by andreasjacobsen93** [comment]
I think we should explore more how deck and tasks can be integrated? I think it could result in some really valuable features.

**Comment by poVoq** [comment]
Any update on the CalDAV tasks and calendar integrations?

**Comment by e-alfred** [comment]
Would a Deck webdav endpoint be automatically added to the Calendar/Tasks app or do users have to manually add them?

**Comment by juliushaertl** [comment]
> ~nextcloud/server#2829~  will allow apps to register custom sabre plugins, if this is in we could create our own CalDAV endpoint at /dav/deck that will allow users to subscribe to within their calendar apps.

I've just updated the PR to https://github.com/nextcloud/server/pull/6835 to get this moving again.

> Would a Deck webdav endpoint be automatically added to the Calendar/Tasks app or do users have to manually add them?

The only issue I see after digging a bit into the sabredav/Nextcloud dav code is, that we cannot easily add the boards to the list of a users calendars. Maybe we need some API for that on the server side. Without that you would need to add it manually.

**Comment by juliushaertl** [comment]
I took some time to dig into the Nextcloud dav app. Here are some notes about the planned implementation. Feedback is very welcome.

## Implementation

- Extend classes from \OCA\DAV\CalDAV (CalendarRoot, Calendar, Home, Calendar, CalendarObject, CalDavBackend)
- With https://github.com/nextcloud/server/pull/6835 register our own sabre collection

- add a collection at /deck with calendar collections at /deck/user/deck-1
- add VTODO entries for every card there
- Showing them in the calendar is part of the caldav client (Thunderbird already has an option for that) 
- We should have an additional field for every card to store vtodo properties we don't have mapped to card values (and show them in the ui, like PERCENT-COMPLETE, PRIORITY

## Mapping VTODO -> Card

- DUE: duedate
- SUMMARY: title
- DESCRIPTION: text
- CATEGORY: tags
- STATUS: archived/unarchived
- URL: url to the card

### Open issues to discuss

#### calendar-home-set

Calendar collections cannot be added to the principals calendar-home-set. Maybe we need to extend Nextclouds CalendarHome::getChildren so apps can add their own entries to the home collection

#### Stack representation
Since nested calendar collections are not possible (something like deck-1/stack-1/card-1.ics) are not possible, we still need to figure out how we can represent the current stack in the caldav representation:
  - we could just obmit it, users can then access their cards for now but not move them between stacks (that is what i would prefer for now)
  - we could create a calendar collection for each stack ( deck-1-stack-1/ deck-1-stack-2/) but that would lead to a huge amount of calendars that would appear in the caldav clients ui
  - We could use the category property, but that should imo be used for tags.

**Comment by slowhand93** [comment]
First of all: thanks for your work Julius! I love Deck.

Regarding stack representation:
I would prefer your second option because I use Deck to organize my days/weeks. So I often move cards between stacks. When I'm understanding you right, this wouldn't be possible if I will use the caldav feature?!

**Comment by poVoq** [comment]
Hmm... while not perfect, it might be doable to represent stacks via the priority value as usually the stacks are related to some sort of todo, backlog etc. system that is similar to priorities.

**Comment by juliushaertl** [comment]
@poVoq That could be a solution, either that or maybe we abuse the LOCATION field which is a string value so we could at least show the stacks name.

**Comment by juliushaertl** [comment]
> I would prefer your second option because I use Deck to organize my days/weeks. So I often move cards between stacks. When I'm understanding you right, this wouldn't be possible if I will use the caldav feature?! 

Another downside of that is that users would need to setup each stack individually in their calendar software, in case the program doesn't provide proper auto discovery (like thunderbird). I would rather prefer one board -> one calendar.

**Comment by 9662** [comment]
Will Deck calendars be generated/assigned on user demand?

My thinking is that not every deck needs a calendar system (not everyone uses due dates and the like) so the user would ideally have an option to have a calendar associated with a deck or not.

Likewise, if the calendar ↔ deck association could be user-configurable, then it may solve some of the issues mentioned above? E.g., as a user you may simply create one or more calendars in Nextcloud and then tell Deck which calendar to write to. This could be on a Deck-wide, per-deck or even per-card basis. I haven't thought this through but this approach may provide the greatest flexibility with relatively conservative development effort.

**Comment by juliushaertl** [comment]
> Will Deck calendars be generated/assigned on user demand?

That is the current idea. We have a separate calendar for each board.

> E.g., as a user you may simply create one or more calendars in Nextcloud and then tell Deck which calendar to write to. 

That would be much more complicated since we would have both events from deck and events from the user in the calendar. Also we need to update every users calendar who the board has been shared with. So using an individual calendar is a much cleaner and easier solution.

> E.g., as a user you may simply create one or more calendars in Nextcloud and then tell Deck which calendar to write to. 

From the current implementation idea, the calendars would be a separate set of calendars from the default nextcloud ones, so it is easy to just not use it in your calendar client. Also I was thinking about allowing users to disable the dav integration per calendar.

In general the idea is that all tasks are exposed as VTODO entry, so they won't show up in your calendar unless the client supports that. There is an issue for the nextcloud calendar app at https://github.com/nextcloud/calendar/issues/28

**Comment by juliushaertl** [comment]
Also an update on the calendar-home-set, i have a WIP branch for the server here https://github.com/nextcloud/server/tree/dav-allow-multiple-homes 

Support of that depends on the client, as it is alway an issue with CalDAV implementations. :wink: DAVdroid works quite nice, but some clients just use the first entry and ignore others.

**Comment by 9662** [comment]
> Also I was thinking about allowing users to disable the dav integration per calendar.

Is that the same as this?

> the user would ideally have an option to have a calendar associated with a deck or not.

As someone who has about 12 decks currently active (one per department / project), I am a bit concerned about potential overabundance of calendars in the Calendar Nextcloud app. ☺

**Comment by bobbiz** [comment]
hello everybody,
thanks to everybody you are here
Like Nextcloud, Like Calendar, Like Deck.
Idea to add new events to my Calendar($DeckName) only, first. 
So here my alternative(maybe not) vision of implementation (pseudocode):

1. Check/create calendar same name as Deck (untill true, maybe move this to createEventMethode).
2. Need 2 methods: create event in callendar and delete event in calendar.
2.1. Create = function() {
  onButtonClick( 
    new Event( $dueDate = not null, $name = $stackName . ' - ' . $cardName, $defaultNotification);
    lock/disable DueDateInput();
    setCardEventCreatedOnce(showDeleteButton(); lockCreateButton(););)};
2.2. Delete = function() {
  onButtonClick(
   unlock/enable DueDateInput();
   checkEventExist($lockedDueDate, $name);
   deleteIfExist(); flipButtons();)}
3. Write Activity / Log every steps listed before.

Deck will not control that Events at all.
Dont punch me please. 
Need Links: CallDav API; Card Edit/View Methods; adding fields; how get authority to use one app from another in same account. Any help Apprisiated.

**Comment by danicotillas** [comment]
I would like to know if finally there is a way to link the card dates with the Calendar.
Thanks!

**Comment by GustavoMarcante** [comment]
Hi. NextCloud is great. I'm using it as file storage and synchronization with my team of "non tech people" since before the fork and is becoming very stable now (both the software and the people habits to use it).
Applications is a great idea. But, in my point of view (based on my team feedback) having a lot of applications to do a lot of tasks confuses more than help.
Again, in my point of view, better than more applications would be to have them integrated.
Researching an "integrated philosophy", I found https://zenkit.com/.
They have lists, kanban (deck), table, calendar (and also mindmaps) all integrated!
Integration with Google Calendar (that many "non tech people" uses) also would be great.
We have A LOT of tools for software development. I don't think NextCloud could compete in this front. But we have a "blank space" related to an open source integrated set of tools like that, to "non tech people" -- the people that is NOW learning to use NextCloud (I mean, not "Yet Another Redmine" kind of think).
Sorry if I wrote too much, but I really like NextCloud and this integration I think would be great.

**Comment by Mannshoch** [comment]
I would prefer to sort the cards by Category. And Stack as Status.
May a category could be upgraded to a Deck.
In thunderbird it appear as normal Category

If I create a task in Thunderbird and don't add at least one Deck category you could put them in nextcloud/deck to a  kind of "unsorted" deck .

The good point is, if I set a category on more then one deck I'm able to put a task into more then one deck. Sometimes there are tasks that belong to more than one project.

**Comment by Mannshoch** [comment]
Hmm... If I add two Deck Category maybe a drawer which contain the link to this Task on the other Deck would be interesting.

**Comment by laurentiu2** [comment]
Moved here from #751 as requested.

I'd find useful to have the option 'Show in Calendar' under Due Date on Tasks, like in the image below.

![screenshot 2019-01-04 at 13 07 52](https://user-images.githubusercontent.com/29857699/50685551-4c818580-1022-11e9-8892-e67d1b5a5729.png)

The same with Deck. The entry in Calendar might be 'Deck/Board_name/Stack_name/Card_name' and the calendar might be the first one in the list - let's say 'default option'.

![screenshot 2019-01-04 at 13 16 29](https://user-images.githubusercontent.com/29857699/50685757-26101a00-1023-11e9-8002-17d38f0c4962.png)

**Comment by simonjaq** [comment]
What's the status on this?
We are using Deck for project management in our company and the feature to integrate with CalDAV would make it absolutely perfect.
The main issue that we have with Deck is the impossibility to add tasks / cards on the go from mobile phones. 
We used to work with nextcloud tasks and 2do on iPhone synced over CalDAV to be able to quickly add tasks. 
The CalDAV integration could be very simple. For example by linking a stack (e. g. Incoming) with a calendar and by adding all tasks that are added to this calendar to the stack.
How much bounty would be needed to get you working on this soon?

**Comment by juliushaertl** [comment]
@simonjaq Unfortunately there are no real news here. However if you use Nextcloud in a corporate environment and wish to have this feature implemented by Nextcloud GmbH, there is the option for sponsoring feature development work on top of your Nextcloud Enterprise subscription to get your features implemented. Feel free to get in contact via https://nextcloud.com/enterprise/buy/

Bountysource is rather a small plattform to give some additional motivation for community contributors.

**Comment by SzOr** [comment]
> @simonjaq Unfortunately there are no real news here. However if you use Nextcloud in a corporate environment and wish to have this feature implemented by Nextcloud GmbH, there is the option for sponsoring feature development work on top of your Nextcloud Enterprise subscription to get your features implemented. Feel free to get in contact via https://nextcloud.com/enterprise/buy/
> 
> Bountysource is rather a plattform to give some additional motivation for community contributors.

It would be THE BEST for the community, it is really a missing piece in the puzzle...

**Comment by simonjaq** [comment]
> @simonjaq Unfortunately there are no real news here. However if you use Nextcloud in a corporate environment and wish to have this feature implemented by Nextcloud GmbH, there is the option for sponsoring feature development work on top of your Nextcloud Enterprise subscription to get your features implemented. Feel free to get in contact via https://nextcloud.com/enterprise/buy/
> 
> Bountysource is rather a plattform to give some additional motivation for community contributors.

Thanks.
We are a small company and not running Nextcloud Enterprise.
What about hiring an independent developer (or you Julius?) to implement this feature?
As I am not a coder it's hard for me to estimate how much work this would mean. Are we speaking about days or weeks of work?
From a non-coder perspective: Couldn't quite some of the functionality be taken from the Nextcloud Tasks app, which is well integrated with CalDAV?

**Comment by juliushaertl** [comment]
@simonjaq I'm not available for hiring, since I'm already working for Nextcloud. However hiring an independent developer is of course also possible. Maybe searching the freelancer section in the forum is an option for that https://help.nextcloud.com/c/nextcloud-freelancing

> From a non-coder perspective: Couldn't quite some of the functionality be taken from the Nextcloud Tasks app, which is well integrated with CalDAV?

This is a bit different, since tasks just used CalDAV as its main data source while deck has its own source in the database, which needs to be mapped to the existing CalDAV endpoint somehow.

**Comment by baldstard** [comment]
@juliushaertl 
> This is a bit different, since tasks just used CalDAV as its main data source while deck has its own source in the database, which needs to be mapped to the existing CalDAV endpoint somehow.

Any progress here?

And what about integration of Deck into Nextcloud Adnroid plugin? This could be a "starting workaround" until mapping will be done. Or not?

**Comment by sehucke** [comment]
> And what about integration of Deck into Nextcloud Adnroid plugin? This could be a "starting workaround" until mapping will be done. Or not?

This depends on the underlying use cases this feature should support.
1. As a user you want to use an existing (CalDAV-supporting) app to display a task list.
2. As a user you want to use an existing (CalDAV-supporting) calendar app to visualize tasks and due dates.
3. As a user you want to use a mobile device to access / work with your Kanban boards.

IMHO, this is a bit out-of-scope for a file sharing app like Nextcloud is. I would vote for a separate Android app that uses public Deck APIs to realize a mobile-friendly frontend. (As my main use case is (3.) :-) )
I would like to start such a project if I find the time...

**Comment by poVoq** [comment]
@sehucke Quite offtopic, but Kandroid would be a good starting point to extend with Deck support:
https://github.com/andresth/Kandroid

**Comment by putt1ck** [comment]
@sehucke you mean something like https://f-droid.org/uk/packages/it.niedermann.nextcloud.deck/ ?

**Comment by sehucke** [comment]
> @sehucke you mean something like https://f-droid.org/uk/packages/it.niedermann.nextcloud.deck/ ?

Exactly! :-) Shame on me that I overlooked this one...

The direct link to the repo: https://github.com/stefan-niedermann/nextcloud-deck

Many contributors are Nextcloud core team members. Looks very promising.

**Comment by raimund-schluessler** [comment]
Follow-up to the discussion in https://github.com/nextcloud/deck/pull/1319, regarding https://github.com/nextcloud/deck/pull/1319#issuecomment-591388534 and https://github.com/nextcloud/deck/pull/1319#issuecomment-591393001.

I think a possible way to map between Deck and CalDAV entities could be the following:

| Deck | CalDAV | Tasks |
|-|-|-|
| Board | `VCALENDAR` | List |
| Stack | `VTODO` without `RELATED-TO` property | Root-Task |
| Card | `VTODO` with `RELATED-TO` (UUID of the Root-Task / Stack) | Subtask |
| Tags | [`CATEGORIES`](https://tools.ietf.org/html/rfc5545#section-3.8.1.2) or custom implementation (Nextcloud tagging system) | Currently uses `CATEGORIES`, integration with Nextcloud tags would be better, see https://github.com/nextcloud/tasks/issues/89 |
| Assigned users | [`ATTENDEE`](https://tools.ietf.org/html/rfc5545#section-3.8.4.1) or custom implementation | [`ATTENDEE`](https://tools.ietf.org/html/rfc5545#section-3.8.4.1) or custom implementation, see https://github.com/nextcloud/tasks/issues/72 |
| Recurring stacks & cards | [`RRULE`](https://tools.ietf.org/html/rfc5545#section-3.8.5.3) | Recurring tasks & subtasks |
| Attachments | [`ATTACH`](https://tools.ietf.org/html/rfc5545#section-3.8.1.1) or custom implementation | [`ATTACH`](https://tools.ietf.org/html/rfc5545#section-3.8.1.1) or custom implementation |

Tags, assigning users, attachments and recurrence is not supported by Tasks yet either, but it would make much sense to implement these features. And if we see that it is not really possible to implement this with pure CalDAV I would be open to implement it without syncing it via CalDAV and have it only available in the webinterface. This would still be better than not having it at all in Tasks and would not limit Deck.

Also having recurring Stacks & Cards could be an interesting option, given that you can add due dates.

For Subsub(..)-Tasks one could show them as Subsub(..)-Cards in Deck or simply redirect to the Tasks app.

Let me know what you think  @nextcloud/deck @nextcloud/calendar

**Comment by juliushaertl** [comment]
Thanks a lot for this valuable input @raimund-schluessler. I must admit I've not been thinking about using the `RELATED-TO` property for representing stacks, but this is a very nice idea.

Since we already have the possibility coming with Nextcloud 19 to expose calendars as an app, I think it makes most sense to tackle the whole CalDAV integration issue in separated milestones.

1. Expose deck boards as VCALENDAR, try to map as much data from deck to the VTODO (read only access)
2. Allow updating VCALENDAR entries, try to map as much as possible from vtodo to the deck database and store the rest as plain vtodo data, so we can still push it out over CalDAV (write access)
3. Deck data could be moved to the DAV backend. I'm thinking about a similar API as https://github.com/juliushaertl/server/blob/5bf3d1bb384da56adbf205752be8f840aac3b0c5/lib/public/Contacts/IManager.php#L54 here. The frontend/apps would still access it though the rest api and have additional capabilties available in deck only that are not representable using CalDAV, but with this deck could also access existing calendars and their VTODO entries and wrap them into a board. We probably need to do some smart logic here to not end up with a board that has 1000 stacks for each first-level task, but that is something to discuss in detail later I'd say.

Step 3 will probably be the most crucial part here, since that means that we also need to migrate existing data and make sure that everything comes nicely together.

**Comment by PhrozenByte** [comment]
Having a CalDAV integration is crucial for us, so I was thinking a little bit about the integration... In [Oct 2017](https://github.com/nextcloud/deck/issues/15#issuecomment-336733486) you guys were discussing about whether or not categories should be used for stacks. I'd strongly vote for it.

The reason is that you should keep in mind how a client, that isn't aware of a Kanban-style tasks list, would represent these tasks. Just think of the "demo board":

![demo board](https://download.bitgrid.net/nextcloud/deck/screenshots/0.5/deck-notifications.png)

Creating a `VTODO` per card would create the tasks "Write a press release", "Take nice screenshots", "Write a blog" and "Think of a slogan". This is without any doubt what a user would expect. But also creating a `VTODO` per stack would also create the tasks "Todo", "In progress" and "Done". What are these tasks supposed to represent? Those tasks can't be assigned to anyone, they can't ever be completed, because that's not what stacks are for, and they will dangle around in every other CalDAV client.

Using `CATEGORIES` for stacks in contrast is more semantically correct: The card "Write a press release" gets the category "Todo", the cards "Take nice screenshots" and "Write a blog" the category "In progress" and the card "Think of a slogan" the category "Done". The relationship between multiple cards is represented by a category just perfect.

But what about tags? You're right, tags are also a good candidate for categories. Why not simply mix them? We could simply define a prefix like `Stack: ` for `CATEGORIES` representing stacks, so the actual `CATEGORIES` property for the card "Write a press release" would end up being `CATEGORIES:Stack: Todo,Pending`. Any other CalDAV client would show these just fine. Even syncing back works: If one creates a new `VTODO` with the category `Stack: Backlog`, Deck could simply create a new stack "Backlog" and show the new card there. Ordering cards and stacks is beyond CalDAV anyway.

**Comment by Fenrihr** [comment]
I saw that Nextcloud 18 has the "Flow" function with similar characteristics to Deck ... so my question is: will there be advances in this function for Deck?
Because i'm using the Snap version of Nextcloud, the 17.0.5 version for be exact.

**Comment by juliushaertl** [comment]
@Fenrihr Flow is something different, see #1722

**Comment by laurentiu2** [comment]
didn't I moved it once?

https://user-images.githubusercontent.com/29857699/50685757-26101a00-1023-11e9-8002-17d38f0c4962.png

**Comment by animorphcoop** [comment]
Thanks to everyone for continuing the development, we've just made a contribution via Bountysource supporting Deck&Calendar integration!

**Comment by juliushaertl** [comment]
Hey there,

I'd just like to let you know that due to the [ongoing issue with Bountysource](https://help.nextcloud.com/t/dont-use-bountysource-anymore/84943) we're discussing ways to move to a different platform. As it looks right now, Bountysource might claim some of the money that was contributed by the community. We will try our best to prevent that. Historically we – the Nextcloud GmbH employees – simply put the claimed bounties back to other tickets until one was solved by a community member. So the money was always *from the community* and *for the community*. We'll therefore move to a better platform and currently don't post any new bounties until this situation is resolved. The current goal is to move/donate the money to [Nextcloud include](https://nextcloud.com/include/) but details will follow in the [forum post](https://help.nextcloud.com/t/dont-use-bountysource-anymore/84943).

Hope this makes sense. If you have any questions please ask them on the forum. This comment will be copy-pasted to all open issues with bounties.

**Comment by makayabou** [comment]
Will wait for it to send a bounty, then. thanks

**Comment by rfc2822** [comment]
I have just tested CalDAV read access and it works fine with DAVx5!

Just wanted to now that on Android / with the OpenTasks API, task lists can't be set as write-only (yet): https://github.com/dmfs/opentasks/issues/851

So people can edit/modify tasks locally, but the changes can't be synchronized to the server. → Write access over CalDAV would be really nice :smirk:

**Comment by juliushaertl** [comment]
Moved out https://github.com/nextcloud/deck/issues/2399 for the write support so we have an individual ticket to track this.

**Comment by q-wertz** [comment]
Didn't find this in the discussions and not sure if it is worth an issue.
So far all the tasks of a Deck where one is participant are published as ToDo. Wouldn't it be more straight forward to only see the tasks one is assigned to?
It makes it quite confusing to have now dozen of Tasks one is not assigned to.

**Comment by alphashuro** [comment]
Is the bounty for this still open?

**Comment by stefan-niedermann** [comment]
@alphashuro please read https://github.com/nextcloud/deck/issues/15#issuecomment-645919501

TL;DR: **No**, please do not post further bounties on Bountysource for any Nextcloud related issue.

**Comment by juliushaertl** [comment]
Closing this as the first integration part is done and we will track write access in https://github.com/nextcloud/deck/issues/2399

**Comment by devvv4ever** [comment]
> Having a CalDAV integration is crucial for us, so I was thinking a little bit about the integration... In [Oct 2017](https://github.com/nextcloud/deck/issues/15#issuecomment-336733486) you guys were discussing about whether or not categories should be used for stacks. I'd strongly vote for it.
> 
> The reason is that you should keep in mind how a client, that isn't aware of a Kanban-style tasks list, would represent these tasks. Just think of the "demo board":
> 
> ![demo board](https://camo.githubusercontent.com/50bd788ffacee7bc85bd1d81fdf86f82a8d244b844f44320c499029fdc987178/68747470733a2f2f646f776e6c6f61642e626974677269642e6e65742f6e657874636c6f75642f6465636b2f73637265656e73686f74732f302e352f6465636b2d6e6f74696669636174696f6e732e706e67)
> 
> Creating a `VTODO` per card would create the tasks "Write a press release", "Take nice screenshots", "Write a blog" and "Think of a slogan". This is without any doubt what a user would expect. But also creating a `VTODO` per stack would also create the tasks "Todo", "In progress" and "Done". What are these tasks supposed to represent? Those tasks can't be assigned to anyone, they can't ever be completed, because that's not what stacks are for, and they will dangle around in every other CalDAV client.
> 

Just a small addition here: why not considering implementing https://tools.ietf.org/html/rfc5545#section-3.6.3 VJOURNAL for such things? VJOURNAL is part of the CalDAV API and it can be used more straightforward as a note or a "simple" task without the need of an end-date or progress. With the RELATED-TO field they could be assigned to an existing VEVENT or VTODO. We hope to get VJOURNAL synchronization into DAVx5 in the future. We want to get that into a supporting app that can display VTODO and VJOURNAL at the same time.

**Comment by HLFH** [comment]
@devvv4ever Create a feature request?

**Comment by PhrozenByte** [comment]
@HLFH See https://github.com/nextcloud/deck/issues/2399#issuecomment-822433043, it's an implementation detail of this partially implemented feature, i.e. no separate feature request.
